lib/PuppeteerSharp.Tests/Page/Events/ConsoleTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Page.Events
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class ConsoleTests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public async Task ShouldWork()
+        {
+            ConsoleMessage message = null;
+
+            void EventHandler(object sender, ConsoleEventArgs e)
+            {
+                message = e.Message;
+                Page.Console -= EventHandler;
+            }
+
+            Page.Console += EventHandler;
+
+            await Page.EvaluateExpressionAsync("console.log('hello', 5, {foo: 'bar'})");
+
+            Assert.Equal("hello 5 JSHandle@object", message.Text());
+            Assert.Equal(ConsoleType.Log, message.Type);
+
+            Assert.Equal("hello", (await message.Args[0].JsonValue()).ToString());
+            Assert.Equal("5", (await message.Args[1].JsonValue()).ToString());
+
+            // TODO Json Serialization is not exactly the same as in Puppeteer
+            Assert.Equal("{\r\n  \"foo\": \"bar\"\r\n}", (await message.Args[2].JsonValue()).ToString());
```

Comment:
Here we should get a json, object, and in line 30 we should get a 5 as an int.

I think that [ValueFromRemoteObject](https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Helper.cs#L22) should try to parse that as a JSON value. I should succeed parsing strings, numbers, and objects.

Reply:
You are right, I'll change it to use remoteObject.type (https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-RemoteObject) with a switch to handle all the cases, what do you think ?

Reply:
Yeah, sounds good

Reply:
maybe make `JsonValue` return `Task<T>` instead of `Task<object>`?

Reply:
Thinking about this a little bit, the doc says:
>Remote object value in case of primitive values or JSON values (if it was requested).

So if we fall there it's because it's serializable, you can just use JSON.Parse in one line

Reply:
I didn't find a way to make it work for the third assert (with `{foo: "bar"}`) without using generics, and for the second (with "5"), there are no integers in JS, only numbers, so I have to force it to float. You'll see all this in my next commit.